src/utils/DepositUtils.ts

```diff
@@ -55,15 +55,21 @@ export async function queryHistoricalDepositForFill(
   let deposit: DepositWithBlock, cachedDeposit: Deposit | undefined;
   if (cache) {
     cachedDeposit = await getDepositInCache(getDepositKey(fill), cache);
+    if (isDefined(cachedDeposit) && !isDepositFormedCorrectly(cachedDeposit)) {
```

Comment:
Should we also delete it from redis when this happens?

Reply:
I was assuming that setting `cachedDeposit = undefined` was James' way to delete it but if not, then we should explicitly delete it

Reply:
By setting it to undefined here, we eventually re-set the key on line 77

Reply:
Ah good point.